### PR TITLE
v2.0 branch

### PR DESCRIPTION
**Pull request by billsaysthis**
* DEPRECATION NOTICE: The sumologic.logger.js module is DEPRECATED and will be removed in v3.0.0. Docs have been updated to reflect this and no changes, including those mentioned in these release notes, will be ported to it.
* POTENTIAL BREAKING CHANGE: The timestamp format used by this library is not handled properly by the Sumo Logic backend and could cause your logs to be inserted at the wrong times. The timestamp format used is updated to be `yyyy-MM-dd'T'HH:mm:ss*SSSZZZZ` (e.g., `2018-08-20'T'13:20:10*633+0000`).
* NEW CAPABILITY: When sending logs or metrics one at a time, including the new config option `returnPromise` with the value `true` forces the `log` function to return a promise and in this case the `onSuccess` and `onError` handlers are ignored. This DEFAULTS to true, so to keep past behavior you MUST pass the topion as `false`.

### CODE REVIEW

File: src/sumoLogger.js [hunk 1]
```diff
@@ -33,6 +34,7 @@ class SumoLogger {
     setConfig(newConfig) {
         this.config = {
             endpoint: newConfig.endpoint,
+            returnPromise: newConfig.hasOwnProperty('returnPromise') ? newConfig.returnPromise : true,
```
Comment (clementallen) [inline src/sumoLogger.js]: Is using `hasOwnProperty` necessary as all the other config items don't use it?

File: src/sumoLogger.js [hunk 2]
```diff
@@ -88,17 +93,26 @@ class SumoLogger {
...
-                { headers }
-            ).then(() => {
-                logsToSend = [];
-                this.config.onSuccess();
-            }).catch((error) => {
-                this.config.onError(error);
-                this.pendingLogs = logsToSend;
-            });
+            if (this.config.returnPromise && logsToSend.length === 1) {
+                const post = axios.post(
```
Comment (clementallen) [inline src/sumoLogger.js]: Can just do `return axios.post(.......`

File: test/sumoLogger.test.js [hunk 3]
```diff
@@ -280,12 +323,13 @@ describe('sumoLogger', () => {
             }, 10);
         });
 
-        it('should call the onError callback if an error object is returned', (done) => {
+        it('should call the onError callback if an error object is returne and returnPromise is falsed', (done) => {
```
Comment (clementallen) [inline test/sumoLogger.test.js]: spelling of returned

File: README.md [hunk 4]
```diff
@@ -57,37 +55,25 @@ $ npm install --save sumo-logger
 **From GitHub:**
 * Download or clone this repo.
 * Copy the files in the `src` folder into your app/website source.
-* Add `<script src="path/to/sumologic.logger.min.js"></source>` or
-   `<script src="path/to/sumologic.logger.js"></source>` to your pages.
-* Add a `<script>` block with the desired log call, as explained in [Usage](#user-content-usage), to your pages.
+* Add `<script src="path/to/sumologic.logger.js"></source>` to your pages or `import` or `require` the module in your app.
```
Comment (clementallen) [inline README.md]: This appears to still encourage use of the old `sumologic.logger.js` library, is it possible to remove?

File: README.md [hunk 5]
```diff
@@ -304,8 +246,8 @@ Please file issues or feature requests on this Github repo.
 
 ## Credits
 
-Thanks to [Clement Allen](https://github.com/clementallen) for his contributions to this project.
+Thanks to [Clement Allen](https://github.com/clementallen) for his many contributions to this project.
```
Comment (clementallen) [inline README.md]: Very kind! 😃

File: ReleaseNotes.md [hunk 6]
```diff
@@ -1,5 +1,10 @@
 # Sumo Logic JavaScript Logging SDK Release Notes
 
+### v2.0.0
+* DEPRECATION NOTICE: The sumologic.logger.js module is DEPRECATED and will be removed in v3.0.0. Docs have been updated to reflect this and no changes, including those mentioned in these release notes, will be ported to it.
+* POTENTIAL BREAKING CHANGE: The timestamp format used by this library is not handled properly by the Sumo Logic backend and could cause your logs to be inserted at the wrong times. The timestamp format used is updated to be `yyyy-MM-dd'T'HH:mm:ss*SSSZZZZ` (e.g., `2018-08-20'T'13:20:10*633+0000`).
+* NEW CAPABILITY: When sending logs or metrics one at a time, including the new config option `returnPromise` with the value `true` forces the `log` function to return a promise and in this case the `onSuccess` and `onError` handlers are ignored. This DEFAULTS to true, so to keep past behavior you MUST pass the topion as `false`.
```
Comment (clementallen) [inline ReleaseNotes.md]: spelling of option

File: README.md (hunk 4 above)
Comment (billsaysthis) [reply README.md]: good catch

File: src/sumoLogger.js (hunk 1 above)
Comment (clementallen) [reply src/sumoLogger.js]: Yup you're completely right!

File: ReleaseNotes.md [hunk 7]
```diff
@@ -1,5 +1,10 @@
 # Sumo Logic JavaScript Logging SDK Release Notes
 
+### v2.0.0
+* DEPRECATION NOTICE: The sumologic.logger.js module is DEPRECATED and will be removed in v3.0.0. Docs have been updated to reflect this and no changes, including those mentioned in these release notes, will be ported to it.
+* POTENTIAL BREAKING CHANGE: The timestamp format used by this library is not handled properly by the Sumo Logic backend and could cause your logs to be inserted at the wrong times. The timestamp format used is updated to be `yyyy-MM-dd'T'HH:mm:ss*SSSZZZZ` (e.g., `2018-08-20'T'13:20:10*633+0000`).
+* NEW CAPABILITY: When sending logs or metrics one at a time, including the new config option `returnPromise` with the value `true` forces the `log` function to return a promise and in this case the `onSuccess` and `onError` handlers are ignored. This DEFAULTS to true, so to keep past behavior you MUST pass the opion as `false`.
```
Comment (clementallen) [inline ReleaseNotes.md]: Nearly there!  `opion` :smile:

**Comment by clementallen** [comment]
Build still failing on some linting error

**Comment by billsaysthis** [comment]
@clementallen the Lint issue is very weird, comes from the eslint configuration. One of the other UI devs is helping me look into it.

**Pull request merged**